submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/util/SubmissionFilterSpecification.kt

```diff
@@ -0,0 +1,57 @@
+package ac.uk.ebi.biostd.persistence.util
+
+import ac.uk.ebi.biostd.persistence.model.Submission
+import ac.uk.ebi.biostd.persistence.model.User
+import org.springframework.data.jpa.domain.Specification
+
+class SubmissionFilterSpecification(userId: Long, filter: SubmissionFilter) {
+    var specification: Specification<Submission>
+
+    init {
+        specification = Specification.where(
+            withUser(userId)).and(withVersionGreaterThan(0))
+        filter.accNo?.let { specification = specification.and(withAccession(it)) }
+        filter.rTimeFrom?.let { specification = specification.and(withFrom(it)) }
+        filter.rTimeTo?.let { specification = specification.and(withTo(it)) }
+        filter.keywords?.let {
```

Comment:
here we can use extension function `applyIfNotBlank` so it looks way more clear.
```
 filter.keywords?.applyIfNotBlank { specification = ...
```